lib/active_merchant/billing/gateways/sigma.rb

```diff
@@ -0,0 +1,235 @@
+require 'nokogiri'
+
+module ActiveMerchant #:nodoc:
+  module Billing #:nodoc:
+    class SigmaGateway < Gateway
+      self.test_url =
+        'http://tokenv2.test.merchantprocess.net/TokenWebService.asmx'
+      self.live_url =
+        'https://gateway.merchantprocess.net/tokenV2/TokenWebService.asmx'
+
+      self.supported_countries = ['US']
+      self.default_currency = 'USD'
+      self.supported_cardtypes = %i[visa master american_express discover]
+
+      self.homepage_url = 'https://sigmamerchantprocessing.com/'
+      self.display_name = 'Sigma - Neo Gateway'
+
+      STANDARD_ERROR_CODE_MAPPING = {}
+
+      def initialize(options = {})
+        requires!(options, :api_key, :merchant_id, :terminal_id)
+        super
+      end
+
+      def store(credit_card, options = {})
+        post = {}
+        add_customer_data('store', post, options)
+        add_payment(post, credit_card)
+
+        commit('token', { CreateToken: post })
+      end
+
+      def purchase(money, payment, options = {})
+        sale = {}
+        add_card_info(sale, payment, options)
+        add_customer_data('purchase', sale, options)
+        add_invoice(sale, money, options)
+        add_shipping_data(sale, options)
+        add_billing_data(sale, options)
+        add_additional_data(sale, options)
+
+        post = { Sale: sale }
+
+        commit('sale', post)
+      end
+
+      def supports_scrubbing?
+        true
+      end
+
+      def scrub(transcript)
+        transcript
+          .gsub(%r{(<tem:CardNumber>).+(</tem:CardNumber>)}, '\1[FILTERED]\2')
+          .gsub(%r{(<tem:APIKey>).+(</tem:APIKey>)}, '\1[FILTERED]\2')
+          .gsub(%r{(<tem:CVV>).+(</tem:CVV>)}i, '\1[FILTERED]\2')
+      end
+
+      private
+
+      def add_card_info(post, payment, options)
+        post.update(
+          cvv: payment.verification_value,
+          accountToken:
+            options[:card_token].presence ||
+              store(payment, options).authorization
+        )
+      end
+
+      def add_customer_data(action, post, options)
+        post[:APIKey] = @options[:api_key]
+
+        if action == 'purchase'
+          post.update(
+            merchantAccountNumber: @options[:merchant_id],
+            terminalName: @options[:terminal_id],
+            emailAddress: options[:email],
```

Comment:
Still mixing `@options` and `options`...

Reply:
ahh yes, sorry, I left this one opened because it's a common pattern through every Gateway implementation (I tried to explain in this comment https://github.com/waysact/active_merchant/pull/191#discussion_r779144315 😊 ) and I'm not sure how we should get rid of this!

Reply:
@jscheid could I get a spin on this ☝️